conda_forge_tick/migrators/replacement.py

```diff
@@ -147,3 +149,57 @@ def migrator_uid(self, attrs: "AttrsTypedDict") -> "MigrationUidTypedDict":
         n = super().migrator_uid(attrs)
         n["name"] = self.name
         return n
+
+
+class MiniReplacement(MiniMigrator):
+    """Minimigrator for replacing one package with another.
+
+    Parameters
+    ----------
+    old_pkg : str
+        The package to be replaced.
+    new_pkg : str
+        The package to replace the `old_pkg`.
+    """
+
+    allowed_schema_versions = [0, 1]
+
+    def __init__(
+        self,
+        *,
+        old_pkg: "PackageName",
+        new_pkg: "PackageName",
+    ):
+        super().__init__()
+        self.old_pkg = old_pkg
+        self.new_pkg = new_pkg
+        self.pattern = re.compile(r"\s*-\s*(%s)(\s+|$)" % old_pkg)
+        self.packages = {old_pkg}
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        requirements = attrs.get("requirements", {})
+        # TODO: do we want to limit to host requirements, like the split
+        # minimigrators did?
```

Comment:
We should make this an option and then turn it on as needed to match the current behavior. For ABI migrations, we only ever want to edit host.